src/compiler/types.ts

```diff
@@ -1685,7 +1685,7 @@ namespace ts {
     export interface ElementAccessExpression extends MemberExpression {
         kind: SyntaxKind.ElementAccessExpression;
         expression: LeftHandSideExpression;
-        argumentExpression?: Expression;
+        argumentExpression: Expression;

```

Comment:
`argumentExpression` can actually be undefined if you have a syntax error like `foo[]`

Reply:
we should create a missing node instead here. almost all the code afterwords does not handel it being undefined.